tests/Unit/ParallelAlgorithms/LinearSolver/Gmres/CMakeLists.txt

```diff
@@ -20,7 +20,9 @@ add_dependencies(
   module_GlobalCache
   )
 
+# Temporarily comment out until the preconditioned elliptic solvers is
+# updated to properly register with observers.
 add_linear_solver_algorithm_test("GmresAlgorithm")
-add_linear_solver_algorithm_test("GmresPreconditionedAlgorithm")
+# add_linear_solver_algorithm_test("GmresPreconditionedAlgorithm")
```

Comment:
Here's the patch to fix these. I can put this in a follow-up PR if you prefer.

```diff
diff --git a/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/DistributedLinearSolverAlgorithmTestHelpers.hpp b/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/DistributedLinearSolverAlgorithmTestHelpers.hpp
index 7c5affe3b..7a444a555 100644
--- a/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/DistributedLinearSolverAlgorithmTestHelpers.hpp
+++ b/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/DistributedLinearSolverAlgorithmTestHelpers.hpp
@@ -328,6 +328,7 @@ struct ElementArray {
           typename Metavariables::Phase,
           Metavariables::Phase::RegisterWithObserver,
           tmpl::list<typename linear_solver::register_element,
+                     helpers::detail::register_preconditioner<preconditioner>,
                      typename linear_solver::prepare_solve,
                      Parallel::Actions::TerminatePhase>>,
       Parallel::PhaseActions<
diff --git a/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/LinearSolverAlgorithmTestHelpers.hpp b/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/LinearSolverAlgorithmTestHelpers.hpp
index b8008cf47..a508d632a 100644
--- a/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/LinearSolverAlgorithmTestHelpers.hpp
+++ b/tests/Unit/Helpers/ParallelAlgorithms/LinearSolver/LinearSolverAlgorithmTestHelpers.hpp
@@ -222,6 +222,20 @@ template <typename Preconditioner>
 using init_preconditioner =
     typename init_preconditioner_impl<Preconditioner>::type;
 
+template <typename Preconditioner>
+struct register_preconditioner_impl {
+  using type = typename Preconditioner::register_element;
+};
+
+template <>
+struct register_preconditioner_impl<void> {
+  using type = tmpl::list<>;
+};
+
+template <typename Preconditioner>
+using register_preconditioner =
+    typename register_preconditioner_impl<Preconditioner>::type;
+
 template <typename Preconditioner>
 struct run_preconditioner_impl {
   using type =
@@ -271,6 +285,7 @@ struct ElementArray {
           typename Metavariables::Phase,
           Metavariables::Phase::RegisterWithObserver,
           tmpl::list<typename linear_solver::register_element,
+                     detail::register_preconditioner<preconditioner>,
                      typename linear_solver::prepare_solve,
                      Parallel::Actions::TerminatePhase>>,
       Parallel::PhaseActions<
```

Reply:
Okay, that seems to work!